packages/search/src/features/search/utils.ts

```diff
@@ -167,3 +201,146 @@ export function queryBuilder(
     }
   }
 }
+
+export function combinationQueryBuilder(
```

Comment:
Can we merge `queryBuilder` and `combinationQueryBuilder` function into one ? Most of the props are same in these 2 functions.

Reply:
Will look into this.